0B.preprocess_plier_pathways.ipynb

```diff
@@ -0,0 +1,489 @@
+       "      <th>55585</th>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>92912</th>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>5373</th>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>5372</th>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4686</th>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "       PID_CASPASE_PATHWAY  PID_P38ALPHABETADOWNSTREAMPATHWAY  \\\n",
+       "55585                    0                                  0   \n",
+       "92912                    0                                  0   \n",
+       "5373                     0                                  0   \n",
+       "5372                     0                                  0   \n",
+       "4686                     0                                  0   \n",
+       "\n",
+       "       REACTOME_APC_C_CDC20_MEDIATED_DEGRADATION_OF_MITOTIC_PROTEINS  \\\n",
+       "55585                                                  0               \n",
+       "92912                                                  0               \n",
+       "5373                                                   0               \n",
+       "5372                                                   0               \n",
+       "4686                                                   0               \n",
+       "\n",
+       "       REACTOME_REGULATION_OF_APOPTOSIS  REACTOME_IRON_UPTAKE_AND_TRANSPORT  \n",
+       "55585                                 0                                   0  \n",
+       "92912                                 0                                   0  \n",
+       "5373                                  0                                   0  \n",
+       "5372                                  0                                   0  \n",
+       "4686                                  0                                   0  "
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "canonical_df['eid'] = [canonical_map[g] for g in canonical_df.index.values]\n",
```

Comment:
https://github.com/greenelab/netreg/blob/5e14ac84de28b722838e533c16cfe7af655258a9/0B.preprocess_plier_pathways.ipynb#L413-L415

Feel free to do however you prefer. I always think method chaining makes a series of operations most readable.

```python
canonical_df = (
    canonical_df
    .assign(eid=canonical_df.index.map(canonical_map))
    .dropna(subset=['eid'])
    .set_index('eid')
)
```